DocAggregatorDotnetCore/TomAggregator.cs

```diff
@@ -38,15 +33,22 @@ public void DoSomeWork()
 
             Directory.CreateDirectory(outputDirectoryPath);
 
-            //Console.WriteLine("New folders to go in: " + outputDirectoryPath);
+            //make array of property refs
+            var ctx = new RiggsTestContext();
+            var listOfDocs = ctx.PropDocs.Select(p => p).ToList();
+            var docsArray = listOfDocs.ToArray();
+
 
-            foreach (var file in fileInfos)
+            //new make directories
+            foreach (var item in docsArray)
             {
+                Console.WriteLine(item.PropRef);
                 //Console.WriteLine(file.Name.Substring(0, offset));
-                var propertyDirectory = Path.Combine(outputDirectoryPath, file.Name.Substring(0, offset));
+                var propertyDirectory = Path.Combine(outputDirectoryPath,item.PropRef.ToString());
```

Comment:
I can see that PropRef is a nullable integer. You should ideally include a null check before using it, or perhaps adopt the listOfDocs query so it does not bring back these if you don't plan on doing anything with those that are null.

Reply:
Fixed!